source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -517,6 +543,36 @@ void OAuth2Filter::finishGetAccessTokenFlow() {
   config_->stats().oauth_success_.inc();
 }
 
+void OAuth2Filter::setCookie(Http::ResponseHeaderMap& headers, const std::string& key,
+                             const std::string& value) const {
+  headers.addReferenceKey(Http::Headers::get().SetCookie, absl::StrCat(key, "=", value));
+}
+
+void OAuth2Filter::setSplitCookie(Http::ResponseHeaderMap& headers, const std::string& key,
+                                  const std::string& data, size_t maxChunkSize = 4000) const {
+  if (data.size() <= maxChunkSize) {
+    // If the data is smaller than the maximum chunk size, no need to split
+    setCookie(headers, key, data);
+    return;
+  }
+
+  if (data.size() > maxChunkSize) {
+    ENVOY_LOG(debug, "cookie size is greater than {}: \n{}", maxChunkSize, key);
```

Comment:
nit: I would prefer to print key in the same line. something like `cookie, {}, size is....`